repo-agent/pkg/llm/command_expansion.go

```diff
@@ -0,0 +1,96 @@
+package llm
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// Regex for finding commands: /namespace:command "args" or /namespace:command args
+// We'll support quoted args with simple quote handling.
+var commandRegex = regexp.MustCompile(`(?m)/([\w-]+):([\w-]+)(?:\s+(".*?"|[^\s]+))?`)
+
+func expandCommands(prompt string, geminiDir string) (string, error) {
+	return replaceCommands(prompt, geminiDir)
+}
+
+func replaceCommands(text string, geminiDir string) (string, error) {
+	// We iterate through all matches and replace them.
+	// Since we might have nested commands (unlikely but possible if a command prompt contains another command?),
+	// we'll just do one pass for now.
+
+	var errs []error
+
+	// FindAllStringSubmatchIndex is better if we want to replace ranges, but ReplaceAllStringFunc is easier.
+	result := commandRegex.ReplaceAllStringFunc(text, func(match string) string {
+		submatches := commandRegex.FindStringSubmatch(match)
+		namespace := submatches[1]
+		command := submatches[2]
+		args := ""
+		if len(submatches) > 3 {
+			args = submatches[3]
+		}
+
+		// Unquote args if quoted
+		if strings.HasPrefix(args, "\"") && strings.HasSuffix(args, "\"") {
+			args = args[1 : len(args)-1]
```

Comment:
What is the expected behavior if the `.toml` file is syntactically incorrect? `os.ReadFile` will succeed, but subsequent parsing might fail. While there's no TOML parsing here (it seems to be treated as plain text), if that ever changes, it would be a failure point. For now, it might be worth adding a comment clarifying that the file is treated as raw text.